lib/board.ex

```diff
@@ -66,28 +47,82 @@ defmodule Board do
     end
   end
 
+  def board_full?(board) do
+    board
+    |> Enum.reject(&mark?/1)
+    |> Enum.empty?
+  end
+
+  def dimension(board) do
+    length(board)
+    |> :math.sqrt
+    |> round
+  end
+
+  def current_lines(board) do
+    [rows(board), columns(board), diagonals(board)]
+    |> List.flatten
+    |> Enum.chunk(dimension(board))
+  end
+
+  def rows(board), do: Enum.chunk(board, dimension(board))
+
+  defp columns(board) do
+    board
+    |> rows
+    |> transpose
+  end
+
+  defp diagonals(board), do: diagonals(board, dimension(board))
+  defp diagonals(board, dimension) do
+    if dimension == 3 do
```

Comment:
This if check feels redundant. 
I'd suggest to either do some math, or pattern match into the 3 and the 4 case.